src/coreclr/tests/src/JIT/Regression/JitBlue/Runtime_34170/Runtime_34170.cs

```diff
@@ -0,0 +1,102 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+[StructLayout(LayoutKind.Explicit)]
+internal struct FloatNonAlignedFieldWithSmallOffset
+{
+    [FieldOffset(1)]
+    public float field;
+
+    public FloatNonAlignedFieldWithSmallOffset(float a)
+    {
+        field = a;
+    }
+}
+
+[StructLayout(LayoutKind.Explicit)]
+internal struct FloatNonAlignedFieldWithLargeOffset
+{
+    [FieldOffset(1021)]
+    public float field;
+
+    public FloatNonAlignedFieldWithLargeOffset(float a)
+    {
+        field = a;
+    }
+}
+
+[StructLayout(LayoutKind.Explicit)]
+internal struct DoubleNonAlignedFieldWithSmallOffset
+{
+    [FieldOffset(1)]
+    public double field;
+
+    public DoubleNonAlignedFieldWithSmallOffset(float a)
+    {
+        field = a;
+    }
+}
+
+[StructLayout(LayoutKind.Explicit)]
+internal struct DoubleNonAlignedFieldWithLargeOffset
+{
+    [FieldOffset(1021)]
+    public double field;
+
+    public DoubleNonAlignedFieldWithLargeOffset(float a)
+    {
+        field = a;
+    }
+}
+
+struct SimpleStruct
+{
+    // the field is aligned inside SimpleStruct.
+    public float field;
+}
+
+[StructLayout(LayoutKind.Explicit)]
+internal struct StructNonAlignedField
+{
+    // SimpleStruct is unaligned, so the result offset to its float field is unaligned.
+    [FieldOffset(1)]
```

Comment:
This access was failing before the PR that introduced #34990 because morph was generating `LCL_FLD` for it.